rpc/server.go

```diff
@@ -141,20 +139,22 @@ func (s *Server) ServeCodec(codec ServerCodec) {
 	s.codecsMu.Lock()
 	if atomic.LoadInt32(&s.run) != 1 { // server stopped
 		s.codecsMu.Unlock()
-		return
+		return &shutdownError{}
 	}
 	s.codecs.Add(codec)
```

Comment:
Edit: Ok, so defer does Close it, but I think it should also remove the codec from the saved set, otherwise it will leak memory.